src/workers/postBoostCanceledCores.ts

```diff
@@ -0,0 +1,78 @@
+import { randomUUID } from 'node:crypto';
+import { systemUser } from '../common';
+import { transferCores, usdToCores } from '../common/njord';
+import { Post } from '../entity';
+import {
+  UserTransaction,
+  UserTransactionProcessor,
+  UserTransactionStatus,
+  UserTransactionType,
+} from '../entity/user/UserTransaction';
+import { TypedWorker } from './worker';
+
+const worker: TypedWorker<'api.v1.post-boost-canceled'> = {
+  subscription: 'api.post-boost-canceled-cores',
+  handler: async (message, con, logger): Promise<void> => {
+    const { data } = message;
+    const { userId, postId, refundAmountUsd, campaignId } = data;
+
+    try {
+      const { transactionId } = await con.transaction(async (entityManager) => {
+        const post = await entityManager
+          .getRepository(Post)
+          .findOneBy({ id: postId });
+
+        if (!post) {
+          return {};
+        }
+
+        const userTransaction = await entityManager
+          .getRepository(UserTransaction)
+          .save(
+            entityManager.getRepository(UserTransaction).create({
+              id: randomUUID(),
```

Comment:
Can we check here if there is already UserTransaction for cancellation refund, technically message could fail or be executed twice and then we would return the Cores to the user twice. 

Maybe if campaignId is UUID we can just use that as transactioId which will automatically dedup because njord will not allow duplicate transaction.

Reply:
We can use the campaign id on creation, but then that would mean we won't be able to use that here during refund.